### PR TITLE
Added documents for application access policy cmdlets

### PR DESCRIPTION
**Pull request by frankpeng7**
Draft documents pending PM's contribution

### CODE REVIEW

File: skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+### Assign an application access poicy to a user
+
+```
+PS C:\> Grant-CsApplicationAccessPolicy -Identity "dc17674c-81d9-4adb-bfb2-8f6a442e4624" -PolicyName "ASimplePolicy"
+```
+
+The command shown above assigns the per-user application access policy "ASimplePolicy" to the user with obect ID "dc17674c-81d9-4adb-bfb2-8f6a442e4624".
+
+### UnAssign an application access poicy from a user
```
Comment (dariomws) [inline skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md:41]:
```suggestion
### Unassign an application access policy from a user
```

File: skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+```
+PS C:\> Grant-CsApplicationAccessPolicy -Identity "dc17674c-81d9-4adb-bfb2-8f6a442e4624" -PolicyName $Null
+```
+
+In the command shown above , any per-user application access policy previously assigned to the user with user (object) ID "dc17674c-81d9-4adb-bfb2-8f6a442e4624" is unassigned from that user; as a result, applications configured in the policy can no longer access online meetings on behalf of that user. To unassign a per-user policy, set the PolicyName to a null value ($Null).
```
Comment (dariomws) [inline skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md:47]:
```suggestion
In the command shown above, any per-user application access policy previously assigned to the user with user (object) ID "dc17674c-81d9-4adb-bfb2-8f6a442e4624" is unassigned from that user; as a result, applications configured in the policy can no longer access online meetings on behalf of that user. To unassign a per-user policy, set the PolicyName to a null value ($Null).
```

File: skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+PS C:\> Grant-CsApplicationAccessPolicy -PolicyName "ASimplePolicy" -Global
+```
+
+The command shown above assigns the per-user application access policy "ASimplePoicy" to all the users in the tenant, except any that have an explicit policy assignment.
+
+## PARAMETERS
+
+### -Identity
+
+Indicates the user (object) ID of the user account to be assigned the per-user online voice routing policy.
```
Comment (dariomws) [inline skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md:69]:
```suggestion
Indicates the user (object) ID of the user account to be assigned the per-user application access policy. 
```

File: skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+Required: True
+Position: 0
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -PolicyName
+
+<TODO: description for parameter?> "Name" of the policy to be assigned. The PolicyName is simply the policy Identity minus the policy scope (the "tag:" prefix). For example, a policy with the Identity tag:ASimplePolicy has a PolicyName equal to ASimplePolicy.
```
Comment (dariomws) [inline skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md:85]:
```suggestion
Name of the policy to be assigned. The PolicyName is simply the policy Identity minus the policy scope (the "tag:" prefix). For example, a policy with the Identity tag:ASimplePolicy has a PolicyName equal to ASimplePolicy.
```

**Comment by get-itips** [comment]
@yogkumgit  as the state of the documents is draft, will review when they become ready.

File: skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+```yaml
+Type: PSListModifier
```
Comment (dariomws) [inline skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md:88]:
```suggestion
Type: String
```

File: skype/skype-ps/skype/New-CsApplicationAccessPolicy.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+author: frankpeng7
+ms.author: frpeng
+ms.reviewer:
+---
+
+# New-CsApplicationAccessPolicy
+
+## SYNOPSIS
+
+Creates a new application access policy. Application access policy contains a list of application(client) IDs. When granted to a user, those applications will be authorized to access online meetings on behalf of that user.
```
Comment (dariomws) [inline skype/skype-ps/skype/New-CsApplicationAccessPolicy.md:17]:
```suggestion
Creates a new application access policy. Application access policy contains a list of application (client) IDs. When granted to a user, those applications will be authorized to access online meetings on behalf of that user.
```

File: skype/skype-ps/skype/New-CsApplicationAccessPolicy.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+### Identity
+
+```
+New-CsApplicationAccessPolicy [-Identity <XdsIdentity>] [-AppIds <PSListModifier>] [-Description <String>]
+```
+
+## DESCRIPTION
+
+<TODO: description>
```
Comment (dariomws) [inline skype/skype-ps/skype/New-CsApplicationAccessPolicy.md:29]:
```suggestion
This cmdlet creates a new application access policy. Application access policy contains a list of application (client) IDs. When granted to a user, those applications will be authorized to access online meetings on behalf of that user.
```

File: skype/skype-ps/skype/New-CsApplicationAccessPolicy.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+
+## PARAMETERS
+
+### -Identity
+
+Unique identifier assigned to the policy when it was created.
+
+```yaml
+Type: XdsIdentity
+Parameter Sets: Identity
```
Comment (dariomws) [inline skype/skype-ps/skype/New-CsApplicationAccessPolicy.md:58]:
```suggestion
Parameter Sets: (All)
```

File: skype/skype-ps/skype/Get-CsApplicationAccessPolicy.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+## PARAMETERS
+
+### -Identity
+
+Unique identifier assigned to the policy when it was created.
+
+```yaml
+Type: XdsIdentity
+Parameter Sets: Identity
```
Comment (dariomws) [inline skype/skype-ps/skype/Get-CsApplicationAccessPolicy.md:58]:
```suggestion
Parameter Sets: (All)
```

File: skype/skype-ps/skype/New-CsApplicationAccessPolicy.md [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+Required: True
+Position: 1
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -AppIds
+
+<TODO: description for parameter?> A list of application (client) IDs. For details of application (client) ID, refer to: [Get tenant and app ID values for signing in](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in).
```
Comment (dariomws) [inline skype/skype-ps/skype/New-CsApplicationAccessPolicy.md:70]:
```suggestion
A list of application (client) IDs. For details of application (client) ID, refer to: [Get tenant and app ID values for signing in](https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in).
```

File: skype/skype-ps/skype/New-CsApplicationAccessPolicy.md [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -Description
+
+<TODO: description for parameter> Specifies the description of the policy.
```
Comment (dariomws) [inline skype/skype-ps/skype/New-CsApplicationAccessPolicy.md:86]:
```suggestion
Specifies the description of the policy.
```

File: skype/skype-ps/skype/New-CsApplicationAccessPolicy.md [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+
+```yaml
+Type: string
```
Comment (dariomws) [inline skype/skype-ps/skype/New-CsApplicationAccessPolicy.md:89]:
```suggestion
Type: String
```

File: skype/skype-ps/skype/Remove-CsApplicationAccessPolicy.md [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+
+### Identity
+
+```
+Remove-CsApplicationAccessPolicy [-Identity <XdsIdentity>]
+```
+
+## DESCRIPTION
+
+<TODO: description>
```
Comment (dariomws) [inline skype/skype-ps/skype/Remove-CsApplicationAccessPolicy.md:29]:
```suggestion
This cmdlet deletes an existing application access policy.
```

File: skype/skype-ps/skype/Remove-CsApplicationAccessPolicy.md [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+
+## PARAMETERS
+
+### -Identity
+
+Unique identifier assigned to the policy when it was created.
+
+```yaml
+Type: XdsIdentity
+Parameter Sets: Identity
```
Comment (dariomws) [inline skype/skype-ps/skype/Remove-CsApplicationAccessPolicy.md:49]:
```suggestion
Parameter Sets: (All)

File: skype/skype-ps/skype/Set-CsApplicationAccessPolicy.md [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+
+### Identity
+
+```
+Set-CsApplicationAccessPolicy [-Identity <XdsIdentity>] [-AppIds <PSListModifier>]
+```
+
+## DESCRIPTION
+
+<TODO: description>
```
Comment (dariomws) [inline skype/skype-ps/skype/Set-CsApplicationAccessPolicy.md:29]:
```suggestion
This cmdlet modifies an existing application access policy.
```

File: skype/skype-ps/skype/Set-CsApplicationAccessPolicy.md [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+
+## EXAMPLES
+
+### Add new app ID to the policy
+
+```
+PS C:\> Set-CsApplicationAccessPolicy -Identity "ASimplePolicy" -AppIds @{Add="5817674c-81d9-4adb-bfb2-8f6a442e4622"}
+```
+
+The command shown above adds an new app ID "5817674c-81d9-4adb-bfb2-8f6a442e4622" to the per-user application access policy ASimplePolicy.
```
Comment (dariomws) [inline skype/skype-ps/skype/Set-CsApplicationAccessPolicy.md:39]:
```suggestion
The command shown above adds a new app ID "5817674c-81d9-4adb-bfb2-8f6a442e4622" to the per-user application access policy ASimplePolicy.
```

File: skype/skype-ps/skype/Set-CsApplicationAccessPolicy.md [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+
+## PARAMETERS
+
+### -Identity
+
+Unique identifier assigned to the policy when it was created.
+
+```yaml
+Type: XdsIdentity
+Parameter Sets: Identity
```
Comment (dariomws) [inline skype/skype-ps/skype/Set-CsApplicationAccessPolicy.md:57]:
```suggestion
Parameter Sets: (All)

File: skype/skype-ps/skype/Set-CsApplicationAccessPolicy.md [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+Required: True
+Position: 1
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -AppIds
+
+<TODO: description for parameter?> A list of application (client) IDs. For details of application (client) ID, refer to: [Get tenant and app ID values for signing in](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in).
```
Comment (dariomws) [inline skype/skype-ps/skype/Set-CsApplicationAccessPolicy.md:69]:
```suggestion
A list of application (client) IDs. For details of application (client) ID, refer to: [Get tenant and app ID values for signing in](https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in).
```

File: skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md [hunk 19]
```diff
@@ -0,0 +1,127 @@
...
+
+## PARAMETERS
+
+### -Identity
+
+Indicates the user (object) ID of the user account to be assigned the per-user online voice routing policy. 
+
+```yaml
+Type: UserIdParameter
+Parameter Sets: Identity
```
Comment (dariomws) [inline skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md:73]:
```suggestion
Parameter Sets: (All)
```

File: skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md [hunk 20]
```diff
@@ -0,0 +1,127 @@
...
+## SYNOPSIS
+
+Assigns a per-user application access policy to one or moer users. After assigning an application access policy to a user, the applications configured in th policy will be authorized to access online meetings on behalf of that user.
+
+## SYNTAX
+
+### FileName
+
+```
+Grant-CsApplicationAccessPolicy [-Identity <UserIdParameter>] [-PolicyName <String>]
```
Comment (dariomws) [inline skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md:24]:
```suggestion
Grant-CsApplicationAccessPolicy [-Identity <UserIdParameter>] [-PolicyName <String>] [-Global]
```

**Review by dariomws — CHANGES_REQUESTED** [review]
Also, I think the following parameters are missing:

- **Get-CsApplicationAccessPolicy**: LocalStore, Tenant, Filter
- **Grant-CsApplicationAccessPolicy**: PassThru, Confirm, Global
- **New-CsApplicationAccessPolicy**: WhatIf, Confirm, Tenant
- **Remove-CsApplicationAccessPolicy**: WhatIf, Confirm, Tenant
- **Set-CsApplicationAccessPolicy**: WhatIf, Confirm, Tenant, Description

**Comment by tseward** [comment]
Ditto!

File: skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md [hunk 21]
```diff
@@ -0,0 +1,126 @@
...
+Accept wildcard characters: False
+```
+
+### -Global
+
+When you use this cmdlet without specifying a user identity, the policy applies to all users in your tenant, except any that have an explicit policy assignment. To skip a warning when you do this operation, specify this parameter.
+
+```yaml
+Type: SwitchParameter
+Parameter Sets: GrantToTenant
```
Comment (get-itips) [inline skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md:105]: This parameter set does not appear in the output of Platy nor the Sintax section, should we put Parameter Sets: (All) ?

File: skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md (hunk 5 above)
Comment (frankpeng7) [reply skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md:88]: I actually believe this should be a list

File: skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md (hunk 21 above)
Comment (frankpeng7) [reply skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md]: updated

File: skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md (hunk 5 above)
Comment (dariomws) [reply skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md:88]: But, is it one policy that you grant to the user? Or multiple policies?

**Review by dariomws — COMMENTED** [review]
> Also, I think the following parameters are missing:
> 
> * **Get-CsApplicationAccessPolicy**: LocalStore, Tenant, Filter
> * **Grant-CsApplicationAccessPolicy**: PassThru, Confirm, Global
> * **New-CsApplicationAccessPolicy**: WhatIf, Confirm, Tenant
> * **Remove-CsApplicationAccessPolicy**: WhatIf, Confirm, Tenant
> * **Set-CsApplicationAccessPolicy**: WhatIf, Confirm, Tenant, Description

We should add these parameters, shouldn't we?

File: skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md (hunk 5 above)
Comment (frankpeng7) [reply skype/skype-ps/skype/Grant-CsApplicationAccessPolicy.md:88]: one policy, but this parameter takes a list (of strings)

**Review by frankpeng7 — COMMENTED** [review]
replying comments

**Comment by frankpeng7** [comment]
@dariomws can you please take a look and help merge this if everything looks good?

**Review by dariomws — APPROVED** [review]
@frankpeng7 I'm ok with this now, I think there are only some missing parameters but maybe you don't want to add them now.

@Kateyanne Please merge this PR. Thanks

**Comment by frankpeng7** [comment]
@dariomws I'm not super familiar with powershell cmdlets so only added needed parameters. Not even sure if we support those "missing" ones. Will probably double check and come back to update this in the future if necessary.

@Kateyanne how long does it take till the document is officially published?

**Comment by Kateyanne** [comment]
Hi Frank,

This should go live late this afternoon.

Thanks,

Kateyanne

**Comment by frankpeng7** [comment]
Hi @Kateyanne I'm still not seeing new docs in here https://docs.microsoft.com/en-us/powershell/module/skype/?view=skype-ps
Do you happen to know if there's anyway to track it?

**Comment by Kateyanne** [comment]
Looking into this, Frank.

**Comment by Kateyanne** [comment]
Frank – could you check now?

**Pull request merged**